node/node.go

```diff
@@ -33,6 +33,53 @@ import (
 	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
 )
 
+const (
+	// InstanceTypeLabelKey is part of the built-in Node labels.
+	// See https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#interlude-built-in-node-labels
+	InstanceTypeLabelKey = "beta.kubernetes.io/instance-type"
+
+	// InstanceTypeLabelValue will clearly brand the VK node for proper parsing
+	// such as required by cloud-providers.
+	// As an example, see https://github.com/virtual-kubelet/virtual-kubelet/issues/692
+	InstanceTypeLabelValue = "virtual-kubelet"
+
+	// OSLabelKey is part of the built-in Node labels.
+	// See https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#interlude-built-in-node-labels
+	OSLabelKey = "beta.kubernetes.io/os"
+)
+
+// NewNodeSpec builds a Kubernetes Node resource.
+func NewNodeSpec(name string, taint *corev1.Taint, version string) *corev1.Node {
+	taints := make([]corev1.Taint, 0)
+
+	if taint != nil {
+		taints = append(taints, *taint)
+	}
+
+	node := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"type":                   "virtual-kubelet",
```

Comment:
In case you're wondering, I didn't want to remove this because someone out there may be using it already.